systems-management/nisysmgmt.yml

```diff
@@ -0,0 +1,2030 @@
+﻿swagger: '2.0'
+info:
+  description: 'Management of distributed systems.'
+  version: '1'
+  title: 'SystemLink Systems Management'
+  contact:
+    name: National Instruments
+    url: https://www.ni.com/systemlink
+    email: support@ni.com
+basePath: '/nisysmgmt'
+x-ni-routing-key: 'Skyline.Salt'
+consumes: [application/json]
+produces: [application/json]
+
+securityDefinitions:
+  basicAuth:
+    type: basic
+  cookieAuth:
+    type: apiKey
+    in: header
+    name: Cookie
+
+security:
+  - basicAuth: []
+  - cookieAuth: []
+
+responses:
+  Error:
+    description: Error
+    schema:
+      description: Error response
+      title: ErrorResponse
+      type: object
+      properties:
+        error:
+          $ref: '#/definitions/Error'
+
+  BadRequest:
+    description: Bad Request indicates that the server was unable to process the request because of invalid syntax.
+    schema:
+      title: Bad Request Error Response
+      description: Bad request error response containing error information.
+      type: object
+      properties:
+        error:
+          $ref: '#/definitions/Error'
+
+  Unauthorized:
+    description: Not authorized.
+    headers:
+      WWW_Authenticate:
+        description: Information for how to authenticate.
+        type: string
+
+  PartialSuccess:
```

Comment:
This partial success response only includes an error object.  What data is included for the items that were successful?  Generally, the partial success response definition needs to be specific to each type of resource / route.

Reply:
For almost all routes I think returning the failed items is enough. The only route that could be addressed is /v1/create-groups which could return the generated gids. I will modify it.

Reply:
The "Responses" section of the SystemLink HTTP API Guidelines indicates 'partial success' responses should contain more than just the error definition.

Reply:
That is right: "If any resources were created, should include a success array of objects in the response". So in this case only the CreateGroups route should contain additional information. The other routes are RPCs so I think they are fine.